FactoryWeather/App/VIP Home/HomeRouter.swift

```diff
@@ -36,7 +32,7 @@ extension HomeRouter: HomeRoutingLogic {
         guard let viewController = viewController else { return }
         let searchViewController = SearchViewController(safeAreaInsets: viewController.view.safeAreaInsets)
         searchViewController.didSelectLocation = { [weak self] weather, location in
-            self?.viewController?.getWeather(forLocation: location)
+            self?.viewController?.getWeather(forLocation: location, completion: { })
```

Comment:
Why don't we pass in `nil` for the `completion` closure here?

Reply:
Interesting, when we pass function as completion `() -> Void` we need to declare that function is escaping, but when we pass `(() -> Void)?` there is no need for `@escaping`.

Reply:
Yep. That's why I almost always use `optional` closure parameters :)